src/tunnel.c

```diff
@@ -584,27 +585,22 @@ static int tcp_connect(struct tunnel *tunnel)
 
 		memset(&(request), '\0', sizeof(request));
 		for (int j = 0; response == NULL; j++) {
-			/*
-			 * Coverity detected a defect:
-			 *  CID 200508: String not null terminated (STRING_NULL)
-			 *
-			 * It is actually a false positive:
-			 * • Function memset() initializes 'request' with '\0'
-			 * • Function read() gets a single char into: request[j]
-			 * • The final '\0' cannot be overwritten because:
-			 *      j < ARRAY_SIZE(request) - 1
-			 */
-			ssize_t bytes_read = read(handle, &(request[j]), 1);
-
-			if (bytes_read < 1) {
+			if (j >= ARRAY_SIZE(request) - 1) {
 				log_error("Proxy response is unexpectedly large and cannot fit in the %lu-bytes buffer.\n",
 				          ARRAY_SIZE(request));
 				goto err_proxy_response;
 			}
 
-			// detect "200"
+			ssize_t bytes_read = read(handle, &(request[j]), 1);
 			static const char HTTP_STATUS_200[] = "200";
 
+			if (bytes_read < 1) {
+				log_error("Proxy response does not contain \"%s\" as expected.\n",
+				          HTTP_STATUS_200);
+				goto err_proxy_response;
+			}
+
+			// detect "200"
 			response = strstr(request, HTTP_STATUS_200);
```

Comment:
I don't understand how you can tell that the HTTP status code 200 has not been found before you do the strstr(). 
Sure, if bytes_read < 1 no character is returned by the read() command, which is a situation which we should catch at this point, but the status code is a different thing which we have to catch later, or maybe I'm missing something.